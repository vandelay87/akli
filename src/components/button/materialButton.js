import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import Ripple from '../../hooks/ripple';
import { robotoRegular } from '../../styles/fonts';
import { color } from '../../styles/colors';

const MaterialButton = ({ value, click, raised }) => (
  <StyledButton onClick={click} modifiers={raised && 'raised'} role="button">
    {value}
    <Ripple color={color.primary} duration={600} />
  </StyledButton>
);

const BUTTON_CONFIG = {
  raised: () => `
    background: ${color.primary};
    color: #fff;
    box-shadow: rgba(0,0,0,0.2) 0px 3px 6px;

    &:hover {
      background: #3a91e7;
      color: #fff;
      box-shadow: rgba(0,0,0,0.4) 0px 3px 8px;
    }

    &:focus {
      background: #3a91e7;
      color: #fff;
      box-shadow: rgba(0,0,0,0.4) 0px 3px 8px;
    }
  `,
};
const StyledButton = styled.button`
  ${robotoRegular}
  transition: background 0.3s ease;
  background: transparent;
  color: ${color.primary};
  min-width: 64px;
  border-radius: 4px;
  border-width: 0;
  border-style: none;
  text-transform: uppercase;
  font-size: 14px;
  padding: 8px;
  overflow: hidden;
  position: relative;

  &:hover {
    background: #f6fafd;
    color: #1976d2;
    cursor: pointer;
  }

  &:focus {
    background: #e5effa;
    outline: none;
  }

  ${applyStyleModifiers(BUTTON_CONFIG)}; 
`;

MaterialButton.propTypes = {
  value: PropTypes.string.isRequired,
  click: PropTypes.func.isRequired,
  raised: PropTypes.bool,
};
MaterialButton.defaultProps = {
  raised: false,
}

export default MaterialButton;
