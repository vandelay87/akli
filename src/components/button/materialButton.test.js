import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import MaterialButton from './materialButton'
import { basicButton, raisedButton } from './materialButton.stories'

describe('materialButton.js', () => {
  it('renders component', () => {
    const basicButtonTree = renderer.create(basicButton()).toJSON()
    const raisedButtonTree = renderer.create(raisedButton()).toJSON()

    expect(basicButtonTree).toMatchSnapshot()
    expect(raisedButtonTree).toMatchSnapshot()
  })

  it('calls function when clicked', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(<MaterialButton value="click" click={mockFn} />)

    wrapper.simulate('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
