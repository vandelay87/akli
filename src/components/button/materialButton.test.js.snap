// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`figureImage.js renders component 1`] = `
<button
  className="sc-bwzfXH fBYzTV"
  onClick={[Function]}
  role="button"
>
  Click me
  <div
    className="sc-bdVaJa hMQdQq"
    color="#1976d2"
    onMouseDown={[Function]}
  />
</button>
`;

exports[`figureImage.js renders component 2`] = `
<button
  className="sc-bwzfXH ePwoXT"
  onClick={[Function]}
  role="button"
>
  Toggle raised
  <div
    className="sc-bdVaJa hMQdQq"
    color="#1976d2"
    onMouseDown={[Function]}
  />
</button>
`;
